code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -171,7 +177,27 @@
 		force = 30
 	..()
 
+/obj/item/tool/shovel/etool/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/weapon/claymore/mercsword/machete))
```

Comment:
Maybe allow every type of sharp tool to be able to sharpen the etool?
I'm thinking about the knife in your boot for example. Just an idea